src/challenges2.js

```diff
@@ -1,23 +1,108 @@
 // Desafio 10
-function techList() {
-  // seu código aqui
+function techList(array, name) {
+  if (array.length === 0) {
+    return 'Vazio!';
+  }
+
+  if (array.length === 5) {
+    let technologies = [];
+    let sortedArray = array.sort();
+    let user = name;
+    // Próximas linhas inspiradas em https://stackoverflow.com/questions/2383484/how-to-create-a-dynamic-object-in-a-loop
+    for (let i = 0; i < array.length; i += 1) {
+      technologies[i] = {
+        tech: sortedArray[i],
+        name: user,
+      };
+    }
+    return technologies;
+  }
 }
 
+// console.log(techList(["React", "Jest", "HTML", "CSS", "JavaScript"], "Lucas"));
+
 // Desafio 11
-function generatePhoneNumber() {
-  // seu código aqui
+
+// Ideia do código abaixo:
+// A função reduce pega cada valor e verifica se ele é igual ao valor parâmetro, se sim, acrescenta 1 ao acumulador
+// https://www.codegrepper.com/code-examples/javascript/javascript+count+number+of+occurrences+in+array
+function countOcurrences(arr, val) {
+  return arr.reduce((a, v) => (v === val ? a + 1 : a), 0);
+}
+
+function returnErrorMessage(array) {
+  for (let value of array) {
+    if (value < 0 || value > 9 || countOcurrences(array, value) >= 3) {
+      return true;
+    }
+  }
+}
+
+function generatePhoneNumber(array) {
+  if (array.length !== 11) {
+    return 'Array com tamanho incorreto.';
+  }
+  let isError = returnErrorMessage(array);
+  if (isError) {
```

Comment:
```suggestion
  if (returnErrorMessage(array)) {
```
Você também pode usar diretamente o resultado da função dentro do `if` sem precisar criar a variável se quiser. :+1:

Reply:
Boa!